File: src/main/java/com/jss/osiris/modules/osiris/tiers/service/CompanySizeService.java
package com.jss.osiris.modules.osiris.tiers.service;

import java.util.List;

import com.jss.osiris.modules.osiris.tiers.model.CompanySize;

public interface CompanySizeService {
    public List<CompanySize> getCompanySizes();

    public CompanySize getCompanySize(Integer id);

    public CompanySize addOrUpdateCompanySize(CompanySize companySize);
}

File: client/src/app/modules/tiers/services/company.size.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { CompanySize } from '../../tiers/model/CompanySize';

@Injectable({
  providedIn: 'root'
})
export class CompanySizeService extends AppRestService<CompanySize>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getCompanySizes() {
    return this.getList(new HttpParams(), "company-sizes");
  }
  
   addOrUpdateCompanySize(companySize: CompanySize) {
    return this.addOrUpdate(new HttpParams(), "company-size", companySize, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
